### PR TITLE
Azure pipeline work

### LINKED ISSUE #93: MacOS Installation: Application cannot run due to unverified developer source
When running the OSx exe on the  Preview 3 release, the application installation is inturrupted by the following error:
Screenshot 2022-03-29 at 14 06 57

### PR DESCRIPTION
**Pull request by calebkiage**
Sign & notarize MacOS binaries. closes #93
Add compliance checks. see #193 

Policheck pending #193

### CODE REVIEW

File: .azure-pipelines/templates/nuget-packages.yaml [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  - pwsh: |
+      Write-Verbose -Verbose "useCache = '${{parameters.useCache}}'"
+      Write-Verbose -Verbose "restorePath = '${{parameters.restorePath}}'"
+      Write-Verbose -Verbose "extraRestoreArgs = '${{parameters.extraRestoreArgs}}'"
+    displayName: Testing parameters
+
+  # - task: UseDotNet@2
+  #   displayName: 'Use .NET 6'
+  #   inputs:
+  #     version: 7.x
```
Comment (baywet) [inline .azure-pipelines/templates/nuget-packages.yaml]:
```suggestion
```

File: .azure-pipelines/release-cli.yaml [hunk 2]
```diff
@@ -0,0 +1,406 @@
...
+    - src
+    - msgraph-cli.sln
+
+pool:
+  name: Azure Pipelines
+  vmImage: ubuntu-latest
+
+variables:
+  - name: repositoryConnection
+    value: 'GitHub - calebkiage'
```
Comment (baywet) [inline .azure-pipelines/release-cli.yaml:20]: Can you use another connection that's not specific to a user to avoid connections duplication?

**Review by baywet — CHANGES_REQUESTED** [review]
Thanks for solving this.
My main concern at this point is there are a lot of inline PowerShell scripts that are going to be harder to maintain than if they were extracted to their own script files. Would you mind adjusting that aspect please?

**Comment by calebkiage** [comment]
> Thanks for solving this. My main concern at this point is there are a lot of inline PowerShell scripts that are going to be harder to maintain than if they were extracted to their own script files. Would you mind adjusting that aspect please?

The scripts aren't reused and are very specific to the signing. I'll try to split them out, but it might take some time to test. Especially with the variable substitution on azure pipelines. Each build takes at least 15 minutes.

File: .azure-pipelines/release-cli.yaml (hunk 2 above)
Comment (calebkiage) [reply .azure-pipelines/release-cli.yaml:20]: Do we already have a connection that can be reused? I can try it.

Comment (baywet) [reply .azure-pipelines/release-cli.yaml:20]: recap from our discussion on team, there's no main connection, we can leave it as is for now

**Comment by calebkiage** [comment]
Work to extract the powershell scripts has been started on #244

**Review by baywet — APPROVED** [review]

**Review by baywet — APPROVED** [review]

**Review by baywet — APPROVED** [review]

**Pull request merged**